external/nginx/src/core/ngx_buf.h

```diff
@@ -152,7 +154,10 @@ ngx_chain_t *ngx_alloc_chain_link(ngx_pool_t *pool);
 
 
 ngx_int_t ngx_output_chain(ngx_output_chain_ctx_t *ctx, ngx_chain_t *in);
+IA2_END_NO_WRAP
+/* ngx_chain_writer is used as a callback, so must be wrapped */
```

Comment:
Callbacks don't necessarily have to be wrapped, it would just run with pkey 0 if it wasn't. Does `ngx_chain_writer` have to have access to nginx's compartment?

Reply:
Yeah, it calls some nginx logging functions, which aren't wrapped because they're in the same library. At least iirc that was the issue. It definitely needed to be wrapped.